compliance_checker/cf/cf.py

```diff
@@ -3225,13 +3225,17 @@ def check_all_features_are_same_type(self, ds):
         feature_types_found = defaultdict(list)
         for name in self._find_geophysical_vars(ds):
             feature = cfutil.guess_feature_type(ds, name)
-            # If we can't figure out the feature type, don't penalize, just
-            # make a note of it in the messages
+            # If we can't figure out the feature type, penalize. Originally, 
+            # it was not penalized. However, this led to the issue that the
+            # message did not appear in the output of compliance checker if
+            # no other error/warning/information was printed out for section
+            # 9.1.
             if feature is not None:
                 feature_types_found[feature].append(name)
             else:
-                all_the_same.messages.append("Unidentifiable feature for variable {}"
-                                             "".format(name))
+                all_the_same.assert_true(False,
```

Comment:
Stylistically, it'd probably be better to do
```python
all_the_same.out_of += 1
if feature is not None:
    all_the_same.score += 1
    feature_types_found[feature].append(name)
else:
    all_the_same.messages.append("Unidentifiable feature for variable {}".format(name))
```
so that passing features get included in the score.  Ultimately, what's important is that `score < out_of` for a check that has issues in some fashion, but if a given check passes a condition, it should usually be counted in the score.  Sorry if my explanation was a bit confusing in #679 -- I thought you meant giving a single fail status if *any* variable was unidentifiable, or a pass otherwise.  This would have been fine as well, although many of the checks in the compliance checker add to the score/out_of for each condition checked.

Reply:
> so that passing features get included in the score.

I overlooked that. Sorry. Maybe this solution is a bit nicer:

```python
found = False
if feature is not None:
    feature_types_found[feature].append(name)
    found = True

all_the_same.assert_true(found, 
                                      "Unidentifiable feature for variable {}"
                                      "".format(name)
```

Reply:
That'd be fine as well.  Throw an updated commit into this PR and I'll squash + merge the changes.